src/application.rs

```diff
@@ -82,11 +82,11 @@ pub trait ApplicationHandler<T: 'static = ()> {
     /// [`Suspended`]: Self::suspended
     fn resumed(&mut self, event_loop: &ActiveEventLoop);
 
-    /// Emitted when an event is sent from [`EventLoopProxy::send_event`].
+    /// Called when user requested wake up via [`wake_up`] occurs.
```

Comment:
That's a lot of words for something that should behave more or less like `RedrawRequested`.

I'd say that the:

1. Order is not defined, but generally should be before `RedrawRequsted`.
2. Events are squashed, as `request_redraw`.

Events being lost or busy can not really happen due to semantics of how it's written or should be written, if it does it's a bug.

Reply:
I'll outline that the semantics must be:

1. It's delivered only once between `NewEvents` and `AboutToWait`, no matter how much you ask for it. That's how e.g. `eventfd` works, for example(and that's what we use on linux here). 
2. It can not _fail_, winit must ensure that you get a wakeup.


Now I also think that we should communicate errors just in case for forward compat/backward, since it's a good way to detect whether the loop is still alive from auxiliary threads.

At least error should have the option of `LoopLost` or something like that.


So as for the current implementation you already can see that everything is following 1, but as for the 2, there's no guarantee for that, so maybe adding an error for now is a good way forward with that.

Reply:
Resolved in https://github.com/rust-windowing/winit/pull/3694/commits/d03176f68519a6c01b865229c997cae404b28fa2.